### PR TITLE
post-process: support run table analyze after all tables are finished 

### PR DESCRIPTION
**Pull request by glorv**
### What problem does this PR solve? 
Add a new config `analyze-at-last` to allow run all table analyze tasks after all tables are finished. This is useful if TiKV instance's cpu is not rich (e.g. in standard dbaas env, TiKV is deployed with 8core cpu), thus the analyze operation will have significant impacts on SST ingest opeartions.

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Side effects

Related changes

 - Need to update the documentation

### CODE REVIEW

**Comment by kennytm** [comment]
how about #358 🤔

**Comment by glorv** [comment]
> how about #358 🤔

Even based on #358, we still need to do extra work to allow delay analyze phase until all other tasks are finished. Since in our current design, post-process are always executed together.

**Comment by kennytm** [comment]
Okay.

File: lightning/restore/restore.go [hunk 1]
```diff
@@ -1230,17 +1270,20 @@ func (t *TableRestore) postProcess(ctx context.Context, rc *RestoreController, c
...
 			if err != nil {
-				return errors.Trace(err)
+				return false, errors.Trace(err)
 			}
 		}
+		cp.Status = CheckpointStatusChecksummed
 	}
 
 	// 5. do table analyze
+	doAnalyze := true
```
Comment (lance6716) [inline lightning/restore/restore.go:1280]: I'm a little confused about `doAnalyze` means this function do or caller should do, hope another name

File: lightning/restore/restore.go [hunk 2]
```diff
@@ -814,7 +825,31 @@ func (rc *RestoreController) restoreTables(ctx context.Context) error {
...
+	default:
+	}
+
+	close(postProcessTaskChan)
+	// otherwise, we should run all tasks in the post-process task chan
+	for i := 0; i < rc.cfg.App.TableConcurrency; i++ {
+		wg.Add(1)
+		go func() {
+			for task := range postProcessTaskChan {
+				_, err := task.tr.postProcess(ctx, rc, task.cp, true)
```
Comment (lance6716) [inline lightning/restore/restore.go:846]: I think the last parameter means "can't delay".  if it's "can delay", I expect that this function may or may not run something in this turn.

Comment (glorv) [reply lightning/restore/restore.go:847]: replaced with a clearer name

**Review by Little-Wallace — APPROVED** [review]
LGTM

**Review by lance6716 — APPROVED** [review]
LGTM

**Pull request merged**